src/snowflake/sqlalchemy/snowdialect.py

```diff
@@ -317,6 +317,29 @@ def get_check_constraints(self, connection, table_name, schema, **kw):
         # check constraints are not supported by Snowflake
         return []
 
+    @reflection.cache
+    def _get_table_primary_keys(self, connection, schema, table_name, **kw):
+        fully_qualified_path = self._denormalize_quote_join(
+            schema, self.denormalize_name(table_name)
+        )
+        result = connection.execute(
+            text(
+                f"SHOW /* sqlalchemy:_get_table_primary_keys */ PRIMARY KEYS IN TABLE {fully_qualified_path}"
+            )
+        )
+        ans = {}
+        for row in result:
+            table_name = self.normalize_name(row._mapping["table_name"])
+            if table_name not in ans:
+                ans[table_name] = {
+                    "constrained_columns": [],
+                    "name": self.normalize_name(row._mapping["constraint_name"]),
+                }
+            ans[table_name]["constrained_columns"].append(
+                self.normalize_name(row._mapping["column_name"])
+            )
+        return ans
```

Comment:
I see the implementation is almost the same as `_get_schema_primary_keys` except for the method signature having `table_name` and sql is using `IN TABLE` + fully qualified path.

let's reuse the `_get_schema_primary_keys` method for table columns, we can use keyword argument to control the logic.

We could do
```python
def _get_schema_primary_keys(self, connection, schema, *, table_name=None, **kw):
    if table_name is not None:  # table name is passed
        fully_qualified_path = self._denormalize_quote_join(
            schema, self.denormalize_name(table_name)
        )
        result = connection.execute(
            text(
                f"SHOW /* sqlalchemy:_get_table_primary_keys */ PRIMARY KEYS IN TABLE {fully_qualified_path}"
            )
        )
    else:  # schema
        result = connection.execute(
            text(
                f"SHOW /* sqlalchemy:_get_schema_primary_keys */PRIMARY KEYS IN SCHEMA {schema}"
            )
        )
        

```